src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs

```diff
@@ -155,9 +155,15 @@ internal void InferParameterBindingSources(ActionModel actionModel)
                 if (bindingSource == null)
                 {
                     bindingSource = InferBindingSourceForParameter(parameter);
+
+                    var originalBindingInfo = parameter.BindingInfo;
+
                     parameter.BindingInfo = new BindingInfo
```

Comment:
It might be better to re-use the original `BindingInfo` instance if it's not null. This way if new properties get added to `BindingInfo`, we don't have to update this `var bindingInfo = parameter.BindingInfo  = parameter.BindingInfo ?? new Bindinfo();`